gflownet/envs/base.py

```diff
@@ -264,6 +308,50 @@ def step(
         _, self.state, action = self._pre_step(action, skip_mask_check)
         return None, None, None
 
+    def step_backwards(
+        self, action: Tuple[int], skip_mask_check: bool = False
+    ) -> Tuple[List[int], Tuple[int], bool]:
+        """
+        Executes a backward step given an action. This generic implementation should
+        work for all discrete environments, as it relies on get_parents(). Continuous
+        environments should re-implement a custom step_backwards(). Despite being valid
+        for any discrete environment, the call to get_parents() may be expensive. Thus,
+        it may be advantageous to re-implement step_backwards() in a more efficient
+        way as well for discrete environments. Especially, because this generic
+        implementation will make two calls to get_parents - once here and one in
+        _pre_step() through the call to get_mask_invalid_actions_backward() if
+        skip_mask_check is True.
```

Comment:
Would it be possible to have some check for continuous environments having implemented this? For example, having separate `DiscreteEnvironment` and `ContinuousEnvironment` classes (from which all envs inherit), and forcing the continuous ones to implement this? In general, this could be a blueprint for what a user needs to implement itself for all of this to work. I am mostly talking from a point of view of someone new not being surprised that the things randomly don't work because they forgot to implement something.

Reply:
Yes, this is in the to-do list indeed.